servers/rendering/rendering_device.h

```diff
@@ -206,6 +206,25 @@ class RenderingDevice : public RenderingDeviceCommons {
 	Error buffer_clear(RID p_buffer, uint32_t p_offset, uint32_t p_size);
 	Vector<uint8_t> buffer_get_data(RID p_buffer, uint32_t p_offset = 0, uint32_t p_size = 0); // This causes stall, only use to retrieve large buffers for saving.
 
+private:
+	/******************/
+	/**** CALLBACK ****/
+	/******************/
+
+public:
+	enum class CallbackResourceType {
+		TEXTURE,
+		BUFFER,
+	};
+
+	struct CallbackResource {
+		CallbackResourceType type = CallbackResourceType::TEXTURE;
+		RID rid;
+		RDG::ResourceUsage usage = RDG::RESOURCE_USAGE_NONE;
```

Comment:
I'm not fully convinced we should be exposing this enum at the RenderingDevice level, as it'll fail to get documented. I think we might need to make an entirely new enumerator and map it to the driver one internally. This enum is entirely for the RenderingDeviceGraph, and we don't want to expose its information to clients of RenderingDevice.

CC @clayjohn would it be acceptable to create a new enum here instead?

Reply:
Is it ok to map them as the same values, so the conversion is free, or do they need to remain stable for the client API?

```cpp
enum CallbackResourceUsage {
    CALLBACK_RESOURCE_USAGE_NONE = RDG::RESOURCE_USAGE_NONE,
    CALLBACK_RESOURCE_USAGE_TEXTURE_SAMPLE = RDG::RESOURCE_USAGE_TEXTURE_SAMPLE,
    // ...  
};
```

Reply:
@DarioSamo I wonder if we should just move the entire `ResourceUsage` enum over from RDG to RDD, so that there is only one copy of it?

Reply:
Final change was to introduce a new enum into `RenderingDevice` that has the same variants as `ResourceUsage` and statically asserts they are the same, so we can cast directly.